src/App.tsx

```diff
@@ -1,18 +1,48 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const total = items.length;
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const numberOfTheLastPage = Math.ceil(total / perPage);
+  const itemsOnTheLasPage = total % perPage || perPage;
+
+  const itemsOnPage = Array<string>(
+    numberOfTheLastPage === currentPage ? itemsOnTheLasPage : perPage,
+  )
+    .fill('Item ')
+    .map((item, index) => (
+      item + (perPage * currentPage - perPage + index + 1)
+    ));
```

Comment:
Could you use `Array.from` ? You could pass callback to it as well

Reply:
done